### DISCUSSION TITLE
Planning of 1.9.0

### DISCUSSION THREAD

**Discussion by georgeef** [moneymanagerex/moneymanagerex#7060] [Development]
v1.9.0 is currently planned in a few days (2025-01-10). This release contains major changes, and since the last released was delayed, there was little time to prepare it.

At least the following is missing:
- [X] #7080
- [X] Remaining tasks in #6860. Nothing critical, but it would be good to finish the implementation of this feature.
- [ ] More time controls in transactions panel (this will make #6860 much more useful).

(~~The second task above is quite straightforward, I estimate a few days.~~ The third task needs more time and test, maybe a few weeks.)

In addition, I would like to implement:
- [X] #7059

~~(I started already and it looks simple, I estimate 1-2 weeks.)~~

## Proposal

- Postpone v1.9.0 by at least a few weeks.
- Implement critical bug fixes in v1.8.x branch and duplicate the fix in master. This is double effort, but it keeps the mainline open for development.

My experience from the previous release, in which mainline was on hold for long time, is that when major features are blocked their development is setback, merging after long time requires considerable effort (and it is more error-prone), and what is most important, developers which to something else and it is difficult to continue from where they stopped. In other words, plans that are realistic as of today, may not be realistic after a long delay.

The mainline should be blocked only briefly before a release (ideally it should never be blocked). Also big PRs should be prioritized. It is much easier to fix a small PR after another big merge, than to fix a big PR after many small merges.

@whalley, @guanlisheng, @n-stein, @vomikan, @renato-mmex as FYI

**Comment by vomikan** [comment]
Deadlines have never been met in the project. These are just tentative dates. So you can safely postpone the deadline.

**Comment by whalley** [comment]
I don't think we should wait any longer than the end of January for the next release, there are a few bugs that need fixing. So I'm OK with pushing out a few weeks.

**Comment by georgeef** [comment]
Thanks, this is more than enough.

Please review and merge intermediate PRs for #7059 (transactions panel for account groups) as soon as possible, otherwise they diverge after other changes and they are quite big to fix them again. It is almost done, I keep it enabled for testing.

**Comment by georgeef** [comment]
Three tasks are done, the fourth (date ranges in transaction panels) will implemented in parts, in a way that does not block the 1.9.0 release.

I close this discussion.
